### PR TITLE
add predicates

### PR DESCRIPTION
**Pull request by longemen3000**
solves #4 . 
the design is overloadable, so if there is a future decision on what predicate is adequate to use with around, it should be easy to add

### CODE REVIEW

File: src/RangeHelpers.jl [hunk 1]
```diff
@@ -56,6 +56,20 @@ function Base.show(io::IO, o::Approach)
...
+
+function preprocess(fx::Base.Fix2{F,T}) where {F,T}
+    d = direction_pred(fx.f)
+    return Approach{T}(fx.x,d)
+end
+
+direction_pred(f::typeof(Base.:<)) = below
+direction_pred(f::typeof(Base.:≤)) = strictbelow
+direction_pred(f::typeof(Base.:>)) = above
+direction_pred(f::typeof(Base.:≥)) = strictabove
```
Comment (jw3126) [inline src/RangeHelpers.jl]:
```suggestion
direction_pred(f::typeof(Base.:<)) = strictbelow
direction_pred(f::typeof(Base.:≤)) = below
direction_pred(f::typeof(Base.:>)) = strictabove
direction_pred(f::typeof(Base.:≥)) = above
```

**Review by jw3126 — COMMENTED** [review]
Awesome, thanks a lot for making this work! I think the aliases are confused, see above. Also, can you add a few tests for `range` that uses this new style and also use ASCII versions like `>=` in the tests in a few places?

**Comment by longemen3000** [comment]
Of course!

**Comment by jw3126** [comment]
Maybe also mention this awesome feature in the README?

File: src/RangeHelpers.jl [hunk 2]
```diff
@@ -56,6 +56,20 @@ function Base.show(io::IO, o::Approach)
...
+preprocess(start,step,stop) = preprocess(start),preprocess(step),preprocess(stop)
+
+preprocess(x) = x
+
+function preprocess(fx::Base.Fix2{F,T}) where {F,T}
+    d = direction_pred(fx.f)
+    return Approach{T}(fx.x,d)
+end
+
+direction_pred(f::typeof(Base.:<)) = strictbelow
```
Comment (aplavin) [inline src/RangeHelpers.jl]: Just `typeof(<)` works, no need to prefix it with Base or :

File: src/RangeHelpers.jl [hunk 3]
```diff
@@ -56,6 +56,20 @@ function Base.show(io::IO, o::Approach)
     print(io, direction_string(direction), "(", o.value, ")")
 end
 
+preprocess(start,step,stop) = preprocess(start),preprocess(step),preprocess(stop)
+
+preprocess(x) = x
+
+function preprocess(fx::Base.Fix2{F,T}) where {F,T}
```
Comment (aplavin) [inline src/RangeHelpers.jl:65]: Also probably `preprocess(fx::Base.Fix2{typeof(==)}) = fx.x`

**Review by jw3126 — APPROVED** [review]
LGTM! Can you bump the version?

**Comment by jw3126** [comment]
Thanks a lot @longemen3000 this is neat. Should I merge?

**Pull request merged**